subsys/net/lib/mqtt_helper/CMakeLists.txt

```diff
@@ -6,8 +6,41 @@
 
 zephyr_library()
 zephyr_library_sources(mqtt_helper.c)
-zephyr_include_directories_ifdef(CONFIG_MQTT_HELPER_PROVISION_CERTIFICATES cert)
 
 if (CONFIG_MQTT_HELPER_PROVISION_CERTIFICATES)
+
 	message(WARNING "Credentials are exposed in non-secure memory. This should be avoided in production.")
+
+	set(APP_CRED_BINARY_DIR ${APPLICATION_BINARY_DIR}/certs)
+	zephyr_include_directories(cert)
+
+	# Function to handle certificate processing
+	function(process_certificate definition_name original_file_name)
+		set(cert_file ${APPLICATION_SOURCE_DIR}/${CONFIG_MQTT_HELPER_CERTIFICATES_FOLDER}/${original_file_name})
+		if(EXISTS ${cert_file})
+			message(WARNING "${original_file_name} found.")
+
+			# Generate the include file name based on the original file name but in the target directory
+			set(inc_file_name "${original_file_name}.inc")
+			string(REPLACE ".pem" "" inc_file_name ${inc_file_name}) # Remove .pem extension
+			string(REPLACE "-" "_" inc_file_name ${inc_file_name}) # Replace dashes with underscores
+			set(inc_file_path ${APP_CRED_BINARY_DIR}/${inc_file_name})
+
+			generate_inc_file_for_target(app ${cert_file} ${inc_file_path})
+
+			# Use the provided definition name directly for the macro
+			set(cert_def ${definition_name})
+			set(${cert_def} "${inc_file_path}" PARENT_SCOPE)
+			add_definitions(-D${cert_def}="${inc_file_path}")
+		endif()
+	endfunction()
+
+	# Explicitly name definitions to match the expected macros in the C source file
+	process_certificate("MQTT_HELPER_CA_CERT" "ca-cert.pem")
+	process_certificate("MQTT_HELPER_CLIENT_CERT" "client-cert.pem")
+	process_certificate("MQTT_HELPER_PRIVATE_KEY" "private-key.pem")
+	process_certificate("MQTT_HELPER_CA_CERT_2" "ca-cert-2.pem")
+	process_certificate("MQTT_HELPER_CLIENT_CERT_2" "client-cert-2.pem")
+	process_certificate("MQTT_HELPER_PRIVATE_KEY_2" "private-key-2.pem")
```

Comment:
```suggestion
```

I think the only use-case is to have an extra CA cert for Azure, so having extra client cert and private key might only cause extra confusion. Perhaps we can remove them until we get a use-case?

Reply:
Ill create a ticket for this since its a bit OOS. I will have to remove handling of those certs from the rest of MQTT helper as well.

Reply:
https://nordicsemi.atlassian.net/browse/CIA-1174